File: data_engineering/pyspark/simple_app/simple_app.py
from pyspark.sql import SparkSession

log_file = "/opt/homebrew/Cellar/apache-spark/3.5.1/README.md"
spark = SparkSession.builder.appName("Simple Application").getOrCreate()
log_data = spark.read.text(log_file).cache()

num_a = log_data.filter(log_data.value.contains("a")).count()
num_b = log_data.filter(log_data.value.contains("b")).count()

print(f"Lines with a: {num_a}, Lines with b: {num_b}")

spark.stop()

File: scala/firstApp/src/main/scala/SimpleApp.scala
/* SimpleApp.scala */
import org.apache.spark.sql.SparkSession

object SimpleApp {
    def main(args: Array[String]): Unit = {
        val logFile = "/opt/homebrew/Cellar/apache-spark/3.5.2/README.md"
        val spark = SparkSession.builder.appName("Simple Application").getOrCreate()
        val logData = spark.read.textFile(logFile).cache()
        val numAs = logData.filter(line => line.contains("a")).count()
        val numBs = logData.filter(line => line.contains("b")).count()
        println(s"Lines with a: $numAs, lines with b: $numBs")
        spark.stop()
    }
}